File: app/src/main/java/dev/ragnarok/fenrir/api/model/server/VkApiUploadServer.java
package dev.ragnarok.fenrir.api.model.server;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class VKApiUploadServer implements Parcelable, UploadServer {

    public static final Creator<VKApiUploadServer> CREATOR = new Creator<VKApiUploadServer>() {
        @Override
        public VKApiUploadServer createFromParcel(Parcel in) {
            return new VKApiUploadServer(in);
        }

        @Override
        public VKApiUploadServer[] newArray(int size) {
            return new VKApiUploadServer[size];
        }
    };
    /**
     * адрес для загрузки фотографий
     */
    @Nullable
    public String upload_url;
    /**
     * идентификатор альбома, в который будет загружена фотография
     */
    public int album_id;
    /**
     * идентификатор пользователя, от чьего имени будет загружено фото
     */
    public int user_id;

    protected VKApiUploadServer(Parcel in) {
        upload_url = in.readString();
        album_id = in.readInt();
        user_id = in.readInt();
    }

    public VKApiUploadServer() {

    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(upload_url);
        dest.writeInt(album_id);
        dest.writeInt(user_id);
    }

    @NonNull
    @Override
    public String toString() {
        return "VKApiUploadServer{" +
                "upload_url='" + upload_url + '\'' +
                ", album_id=" + album_id +
                ", user_id=" + user_id +
                '}';
    }

    @Nullable
    @Override
    public String getUrl() {
        return upload_url;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/api/model/server/VKApiUploadServer.kt
package dev.ragnarok.fenrir.api.model.server

import android.os.Parcel
import android.os.Parcelable
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class VKApiUploadServer : Parcelable, UploadServer {
    /**
     * адрес для загрузки фотографий
     */
    @SerialName("upload_url")
    override var url: String? = null

    /**
     * идентификатор альбома, в который будет загружена фотография
     */
    var album_id = 0

    /**
     * идентификатор пользователя, от чьего имени будет загружено фото
     */
    var user_id = 0L

    internal constructor(parcel: Parcel) {
        url = parcel.readString()
        album_id = parcel.readInt()
        user_id = parcel.readLong()
    }

    @Suppress("UNUSED")
    constructor()

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(url)
        dest.writeInt(album_id)
        dest.writeLong(user_id)
    }

    override fun toString(): String {
        return "VKApiUploadServer{" +
                "upload_url='" + url + '\'' +
                ", album_id=" + album_id +
                ", user_id=" + user_id +
                '}'
    }

    companion object CREATOR : Parcelable.Creator<VKApiUploadServer> {
        override fun createFromParcel(parcel: Parcel): VKApiUploadServer {
            return VKApiUploadServer(parcel)
        }

        override fun newArray(size: Int): Array<VKApiUploadServer?> {
            return arrayOfNulls(size)
        }
    }
}